tools/hangar/__snapshots__/test_corpus/valid/optional_macros.test.w_compile_tf-aws.md

```diff
@@ -0,0 +1,111 @@
+# [optional_macros.test.w](../../../../../examples/tests/valid/optional_macros.test.w) | compile | tf-aws
+
+## inflight.$Closure1-1.cjs
+```cjs
+"use strict";
+const $helpers = require("@winglang/sdk/lib/helpers");
+module.exports = function({ $expect_Util }) {
+  class $Closure1 {
+    constructor({  }) {
+      const $obj = (...args) => this.handle(...args);
+      Object.setPrototypeOf($obj, this);
+      return $obj;
+    }
+    async handle() {
+      const result = ({});
+      (await $expect_Util.equal((result.item === undefined ? undefined : (result.item)?.["id"]), undefined));
```

Comment:
This looks a lot better. I think there's only one issue left -- the part of the expression before the `?.` is emitted twice, so it will get evaluated twice by accident:

```
let makeArray = (): Array<num> => {
  log("hi");
  return [1, 2, 3];
};

log(makeArray()?.contains(2) ?? true); // prints "hi" twice
```

Reply:
added